client/app.jsx

```diff
@@ -8,16 +8,25 @@ class Location extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
+      listingId: this.getListingId(),
       lat: null,
       lng: null,
       city: null,
       state: null,
       country: null,
+      description: null,
+      gettingAround: null,
       showModal: false
     }
     this.toggleModal = this.toggleModal.bind(this);
     
   }
+
+  getListingId(){
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    return urlParams.get('id');
+  }
```

Comment:
Interesting method of getting your url param. Haven't seen it done this way!

Reply:
it works with most modern browsers so I am prob taking a chance.